### PR TITLE
Processor hints doc

### PR DESCRIPTION
**Pull request by gizas**
Documentation update related to enhancement https://github.com/elastic/elastic-agent/issues/2959

### CODE REVIEW

**Review by ChrsMark — APPROVED** [review]

**Review by constanca-m — APPROVED** [review]

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 1]
```diff
@@ -83,6 +83,64 @@ The stream to use for logs collection, for example, stdout/stderr.
 
 If the specified package has no logs support, a generic container's logs input will be used as a fallback. See the `Hints autodiscovery for kubernetes log collection` example below.
 
+[float]
+=== `co.elastic.hints/processors`
+
+Define a processor to be added to the  input configuration. See <<elastic-agent-processor-configuration>> for the list of supported processors.
+
+If processors configuration uses list data structure, object fields must be enumerated. For example, hints for the rename processor configuration below
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:91]:
```suggestion
If the processors configuration uses list data structure, object fields must be enumerated. For example, hints are specified for the rename processor configuration below:
```

Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:91]:
```suggestion
If the processors configuration uses list data structure, object fields must be enumerated. For example, hints for the rename processor configuration below
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 2]
```diff
@@ -83,6 +83,64 @@ The stream to use for logs collection, for example, stdout/stderr.
...
+
+[source,yaml]
+----
+co.elastic.hints/processors.1.dissect.tokenizer: "%{key1} %{key2}"
+co.elastic.hints/processors.dissect.tokenizer: "%{key2} %{key1}"
+----
+
+In the above sample the processor definition tagged with 1 would be executed first.
+
+IMPORTANT: Processor configuration is not supported on datastream level. So annotations like `co.elastic.hints/<datastream>.processors` will be ignored.
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:141]:
```suggestion
IMPORTANT: Processor configuration is not supported on the datastream level, so annotations like `co.elastic.hints/<datastream>.processors` are ignored.
```

**Review by kilfoyle — APPROVED** [review]
LGTM! 👍 
Just a few very small suggestions for the text.

**Comment by gizas** [comment]
@kilfoyle if I want this to be part of the 8.10, what do I have to do? I added the label `8.10-request`. Is that right?

**Comment by kilfoyle** [comment]
if I want this to be part of the 8.10, what do I have to do?

@gizas Normally we can add `backport-8.10.0` label but I see that doesn't exist. Let me try to fix that.

**Comment by kilfoyle** [comment]
Labels are fixed up. :-)

**Comment by gizas** [comment]
@ChrsMark , @constanca-m , @kilfoyle I added another new section in the last commit https://github.com/elastic/ingest-docs/pull/379/commits/2c724774f529fe590848196e1b979ccfca48ebac

Just wanted to notify you because you had already approved it. I keep this PR open as still the main PR https://github.com/elastic/elastic-agent/pull/3107 is not merged due to CI issues

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 3]
```diff
@@ -280,6 +338,46 @@ providers:
     scope: node
 ----
 
+[discrete]
+=== Hints autodiscovery for kubernetes logs with JSON decoding
+
+Based on previous example, users might want to perform extra processing on specific logs, eg. to decode specific fields containing JSON strings. Usage of << decode-json-fields >> is advisable as follows:
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:344]:
```suggestion
Based on the previous example, users might want to perform extra processing on specific logs, for example to decode specific fields containing JSON strings. Use of <<decode-json-fields>> is advisable as follows:
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 4]
```diff
@@ -280,6 +338,46 @@ providers:
...
+
+Users need to have enabled hints autodiscovery as described in `Hints autodiscovery for kubernetes log collection` example before
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:346]:
```suggestion
You need to have enabled hints autodiscovery, as described in the previous `Hints autodiscovery for Kubernetes log collection` example.
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 5]
```diff
@@ -280,6 +338,46 @@ providers:
...
+
+The pod that will produce json logs needs to be annotated with:
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:348]:
```suggestion
The pod that will produce JSON logs needs to be annotated with:
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 6]
```diff
@@ -280,6 +338,46 @@ providers:
...
+
+ annotations:
+        co.elastic.hints/package: "container_logs"
+        co.elastic.hints/processors.decode_json_fields.fields: "message"
+        co.elastic.hints/processors.decode_json_fields.add_error_key: 'true'
+        co.elastic.hints/processors.decode_json_fields.overwrite_keys: 'true'
+        co.elastic.hints/processors.decode_json_fields.target: "team"
+----
+
+> NOTE: Above parameters for decode_json_fields processor are just an example
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:361]:
```suggestion
> NOTE: These parameters for the `decode_json_fields` processor are just an example.
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 7]
```diff
@@ -280,6 +338,46 @@ providers:
...
+
+A following log entry :
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:363]:
```suggestion
The following log entry:
```

File: docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc [hunk 8]
```diff
@@ -280,6 +338,46 @@ providers:
...
+
+[source,json]
+----
+{"myteam": "ole"}
+----
+
+Will produce both fields, the original `message` field and also the target field `team`
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-hints-autodiscover.asciidoc:370]:
```suggestion
Will produce both fields: the original `message` field and also the target field `team`.
```

**Review by kilfoyle — APPROVED** [review]
LGTM! Looks good @gizas!
Just a few small suggestions.

**Pull request merged**